src/unix/fs.c

```diff
@@ -404,6 +404,36 @@ static ssize_t uv__fs_readlink(uv_fs_t* req) {
   return 0;
 }
 
+static ssize_t uv__fs_realpath(uv_fs_t* req) {
+  ssize_t len;
+  char* buf;
+
+  len = pathconf(req->path, _PC_PATH_MAX);
+
+  if (len == -1) {
+#if defined(PATH_MAX)
+    len = PATH_MAX;
+#else
+    len = 4096;
+#endif
+  }
+
+  buf = uv__malloc(len + 1);
+
+  if (buf == NULL) {
+    errno = ENOMEM;
+    return -1;
+  }
+
+  if (realpath(req->path, buf) == NULL) {
```

Comment:
That looks ok to me, Yuval.

On Friday, September 18, 2015, Yuval Brik <notifications@github.com> wrote:

> In src/unix/fs.c
> <https://github.com/libuv/libuv/pull/531#discussion_r39887331>:
>
> > +  if (len == -1) {
> > +#if defined(PATH_MAX)
> > +    len = PATH_MAX;
> > +#else
> > +    len = 4096;
> > +#endif
> > +  }
> > +
> > +  buf = uv__malloc(len + 1);
> > +
> > +  if (buf == NULL) {
> > +    errno = ENOMEM;
> > +    return -1;
> > +  }
> > +
> > +  if (realpath(req->path, buf) == NULL) {
>
> True, realpath() does that, but I thought it would be better to allocate
> using uv__malloc() - for profiling and valgrind. Also, as we freeing that
> memory using its counterpart uv__free().
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/libuv/libuv/pull/531/files#r39887331>.
>


-- 
/Saúl
bettercallsaghul.com